### ISSUE TITLE
New PCB revision

### ISSUE THREAD

**Issue by Spud2233** [absalom-muc/MHI-AC-Ctrl#70]
This is less of an issue and more of a hardware revision.

I have been working on a version of the PCB that is almost fully turn-key,
The fully assembled PCB can be ordered from JLCPCB, PCBway, etc all the end user needs to do is flash the firmware and solder on the header connector or lead.

The fully populated board measures 34.5 x 26.5 with a thickness of 5-6mm without the connectors. 
Instead of using the D1 mini as the basis of the module I have opted for an ESP12F. This may not be ideal for some users as it requires a separate USB to serial adaptor for programing, but I have included a programing header.

Any feedback that you can provide would be appreciated.

![image](https://user-images.githubusercontent.com/9654231/132360573-3dbe5b34-d7d9-4d71-a1cd-4266a99f0566.png)
![image](https://user-images.githubusercontent.com/9654231/132363922-ab207980-db2b-490f-a6ab-f036f491bee1.png)
![image](https://user-images.githubusercontent.com/9654231/132363980-65b1e96a-1fc8-4c5b-834c-c14722311427.png)
![image](https://user-images.githubusercontent.com/9654231/132364069-386327d7-3518-443d-b358-31e438f451df.png)
![image](https://user-images.githubusercontent.com/9654231/132364101-506bd213-3e85-485e-9716-8415d6af641e.png)
![image](https://user-images.githubusercontent.com/9654231/132364180-d7a85d1e-1281-409f-bb2a-c21c791c013c.png)

### **I HAVE NOT BUILT A PROTOTYPE OF THIS YET, USE THE BELOW FILES AT YOUR OWN RISK.**
[BOM_PCB_MHI-AC-Ctrl_v2.8_2021-09-08.csv](https://github.com/absalom-muc/MHI-AC-Ctrl/files/7122527/BOM_PCB_MHI-AC-Ctrl_v2.8_2021-09-08.csv)
[Gerber_PCB_MHI-AC-Ctrl_v2.8_2021-09-08.zip](https://github.com/absalom-muc/MHI-AC-Ctrl/files/7122585/Gerber_PCB_MHI-AC-Ctrl_v2.8_2021-09-08.zip)
[PickAndPlace_PCB_MHI-AC-Ctrl_v2.8_2021-09-08.csv](https://github.com/absalom-muc/MHI-AC-Ctrl/files/7122529/PickAndPlace_PCB_MHI-AC-Ctrl_v2.8_2021-09-08.csv)

**Comment by damienmorrissey** [comment]
Nice!

I was just about to put an order in for some boards. I'll review it and get
back to you.

**Comment by absalom-muc** [comment]
Thank you @Spud2233 for the layout. I had a rough look on the layout with an online Gerber viewer and there might be some problems, but I'm not sure since I have no scale with the Gerber viewer, so I can only estimate sizes and distances:
I have the impression that the wires are very small while you have sufficient area. In addition it seems that the distances between wires and pads are partly too small. example:
![grafik](https://user-images.githubusercontent.com/23119513/132566956-7d9d210f-2368-45d4-9657-a08f2defa21a.png)
The small distances and wire sizes increase the risk for short and opens. In parallel the PCB production costs could be unnecessarily increased.
Not sure if your copper heat sink is sufficient for the heating by the approx. 1 Watt thermal power loss.
You could create a short between the programming connector pins and the ESP12F.

**Comment by Spud2233** [comment]
Thanks for the feedback, I am fairly new to PCB design so all feedback is appreciated.

I have done a bit more research and made a few changes.

To be on the safe side I have Increased clearance between tracks to 0.254mm/10mil, this should be sufficient for up to 30V DC
The power traces  have been increased to 0.381mm/15mil, the calculations said this should be good for 1A, the signal traces are 0.254mm/10mil.
I have added via's and a 5V plane on the back of the PCB to help with heat dissipation from the AMS1117-5, do you think that is sufficient?
I have also added a ground plane to the top layer of the PCB as well as via fencing at the perimeter of the PCB nearest the ESP's antenna, I'm not sure if this is over the top though?

![image](https://user-images.githubusercontent.com/9654231/132704283-5e57b803-612c-4361-81f2-89471e2fffa7.png)
![image](https://user-images.githubusercontent.com/9654231/132704320-2e467aec-9e70-404d-a2ce-8606b693f76d.png)
![image](https://user-images.githubusercontent.com/9654231/132704384-f6a13b01-292d-4d92-ba1e-b3911d69389b.png)
![image](https://user-images.githubusercontent.com/9654231/132704427-89b9d3fb-65a0-49e8-9922-d425bc407e15.png)

### I HAVE NOT BUILT A PROTOTYPE OF THIS YET, USE THE BELOW FILES AT YOUR OWN RISK.
[PickAndPlace_PCB_MHI-AC-Ctrl_v2.9_2021-09-10.csv](https://github.com/absalom-muc/MHI-AC-Ctrl/files/7137360/PickAndPlace_PCB_MHI-AC-Ctrl_v2.9_2021-09-10.csv)
[Gerber_PCB_MHI-AC-Ctrl_v2.9_2021-09-10.zip](https://github.com/absalom-muc/MHI-AC-Ctrl/files/7137361/Gerber_PCB_MHI-AC-Ctrl_v2.9_2021-09-10.zip)
[BOM_PCB_MHI-AC-Ctrl_v2.9_2021-09-10.csv](https://github.com/absalom-muc/MHI-AC-Ctrl/files/7137362/BOM_PCB_MHI-AC-Ctrl_v2.9_2021-09-10.csv)

**Comment by dvisser** [comment]
Good work, however you should really avoid right-angles in your pcb traces and with regards to SMD components, you may get some strange "tombstoning" effects if you don't run tracks in certain directions into SMD pads.
This is the PCB a friend and I designed for this purpose if you want to take a look (still uses wemos module as we like how the usb interface is already there).  

https://github.com/dvisser/MHI-ESP-PCB

**Comment by absalom-muc** [comment]
It is much improved compared to the first version and I believe the board will work.

However, my target is always to maximize the distances and have some buffer for the wire size. e.g.
![grafik](https://user-images.githubusercontent.com/23119513/132869381-7bb487c5-e7ca-4c30-ae94-02b17da3783d.png)
There is no reason having the wire so close to the pin. Why don't you go from the pin to the right and then up?
Might be not a must, but gives you some buffer in case of PCB production issues.
btw: I assume you run a DRC with the ruleset from your PCB vendor?
The ground plane is nice to have, but I believe it is not needed in this case.
And I still see the risk of a short between the programming connector pins and the ESP12F.

**Comment by damienmorrissey** [comment]
Agreed. So much better. I'm just getting into this stuff myself and feel
that extra space can not hurt provided it fits the board. What does it look
like if you increase the minimum distance by another 20%? In some respects
you could almost keep going until the board cannot be resolved within the
space constraints without any harm.

That said the rest of the board looks good so perhaps a manual adjustment
would be fine.

Regarding the additional cost, in my case I'm building 5 of them so
delivered and built it would be just under $50 for all 5. That is excellent
value imo. However, if it's a one off it is not as good of a deal.

**Comment by fonske** [comment]
What are the differences between this board with ESP12F and the previous one with the Wemos mini?

**Comment by absalom-muc** [comment]
closed since there was no update for a long time. Please feel free to re-open in case of follow-up topics.

**Comment by ThomasHFWright** [comment]
@Spud2233 did you ever make an updated board or test these? I'm looking to get some fresh PCBs made and shipped and I'd prefer a turnkey solution if possible.
